src/txorphanage.h

```diff
@@ -43,7 +43,8 @@ class TxOrphanage {
     /** Erase an orphan by wtxid */
     int EraseTx(const uint256& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
 
-    /** Erase all orphans announced by a peer (eg, after that peer disconnects) */
+    /** Maybe erase all orphans announced by a peer (eg, after that peer disconnects). If an orphan
```

Comment:
I think there can be a source of concerns if all the announced orphans are second components from a single common parent. All our peers would announce those second-stage components, only one of them will succeed in the mempool validation and other will be refused as conflicts, I think.

Reply:
Can you elaborate on this concern? The orphan that isn't received yet may still be in orphanage, and will be connected to its parents normally?

Reply:
> Can you elaborate on this concern? The orphan that isn't received yet may still be in orphanage, and will be connected to its parents normally

So if you have 3 peers and they announce A, B, C all with different wtxids while they’re spending the same parent txid, do we have a concern of the parent being fetched from multiple peers, or download should be dedup when we receive the `ancpkginfo`s ?

Reply:
> download should be dedup when we receive the ancpkginfos ?

If it's in your orphanage it won't be fetched again. See `ReceivedAncPkgInfo`